Samples.Objc/Samples.ObjC/API/Requests/CreateOrderRequest.m

```diff
@@ -10,35 +10,99 @@
 
 @implementation CreateOrderRequest
 
-- (id)initWith:(OrderIntent)intent purchaseUnits:(NSArray *)purchaseUnits {
+- (id)initWithOrder:(PPCOrderRequest*)order accessToken:(NSString*)token {
   self = [super init];
   if (self) {
-    self.intent = intent;
-    self.purchaseUnits = purchaseUnits;
+    self.order = order;
+    self.accessToken = token;
   }
   return self;
 }
 
 - (NSDictionary *)properties {
+  NSString *intentString = @"";
+  if (self.order.intent == PPCOrderIntentCapture) {
+    intentString = @"CAPTURE";
+  } else {
+    intentString = @"AUTHORIZE";
+  }
 
-  // Convert purchase units to json
   NSMutableArray *purchaseUnitDictionaries = [NSMutableArray new];
-  for (PurchaseUnit *purchaseUnit in [self purchaseUnits]) {
-    NSDictionary *purchaseUnitDictionary = [purchaseUnit dictionaryRepresentation];
+  for (PPCPurchaseUnit *purchaseUnit in self.order.purchaseUnits) {
+    
+    NSMutableArray *itemsDictionary = [NSMutableArray new];
+    
+    for (PPCPurchaseUnitItem *item in purchaseUnit.items) {
+      
+      NSDictionary *itemUnitAmountDictionary = @{
+        @"currency_code": @"USD",
+        @"value": item.unitAmount.value
+      };
+      
+      NSString *itemCategoryString = nil;
+      if (item.category == PPCPurchaseUnitCategoryDigitalGoods) {
+        itemCategoryString = @"DIGITAL_GOODS";
+      } else {
+        itemCategoryString = @"PHYSICAL_GOODS";
+      }
+      
+      NSDictionary *itemDictionary = @{
+        @"name": item.name,
+        @"unit_amount": itemUnitAmountDictionary,
+        @"quantity": item.quantity,
+        @"category": itemCategoryString
+      };
+      
+      [itemsDictionary addObject:itemDictionary];
+    }
+    
+    NSDictionary *breakdownItemTotalDictionary = @{
+      @"currency_code": @"USD",
+      @"value": purchaseUnit.amount.breakdown.itemTotal.value
+    };
+    
+    NSDictionary *breakdownTaxTotalDictionary = @{
+      @"currency_code": @"USD",
+      @"value": purchaseUnit.amount.breakdown.taxTotal.value
+    };
+    
+    NSDictionary *breakdownDictionary = @{
+      @"item_total": breakdownItemTotalDictionary,
+      @"tax_total": breakdownTaxTotalDictionary
+    };
+    
+    NSDictionary *purchaseUnitAmountDictionary = @{
+      @"currency_code": @"USD",
+      @"value": purchaseUnit.amount.value,
+      @"breakdown": breakdownDictionary
+    };
+    
+    NSDictionary *purchaseUnitDictionary = @{
+      @"amount": purchaseUnitAmountDictionary,
+      @"items": itemsDictionary
+    };
```

Comment:
Something like should suffice I believe, please correct me if I am missing any `nil` checks:

```objectivec
- (NSDictionary *)properties {
  NSString *intentString = @"";
  if (self.order.intent == PPCOrderIntentCapture) {
    intentString = @"CAPTURE";
  } else {
    intentString = @"AUTHORIZE";
  }

  NSMutableArray *purchaseUnitDictionaries = [NSMutableArray new];
  for (PPCPurchaseUnit *purchaseUnit in self.order.purchaseUnits) {
    
    NSMutableArray *itemsDictionary = [NSMutableArray new];
    
    for (PPCPurchaseUnitItem *item in purchaseUnit.items) {
      
      NSDictionary *itemUnitAmountDictionary = @{
        @"currency_code": @"USD",
        @"value": item.unitAmount.value
      };
      
      NSString *itemCategoryString = nil;
      if (item.category == PPCPurchaseUnitCategoryDigitalGoods) {
        itemCategoryString = @"DIGITAL_GOODS";
      } else {
        itemCategoryString = @"PHYSICAL_GOODS";
      }
      
      NSDictionary *itemDictionary = @{
        @"name": item.name,
        @"unit_amount": itemUnitAmountDictionary,
        @"quantity": item.quantity,
        @"category": itemCategoryString
      };
      
      [itemsDictionary addObject:itemDictionary];
    }

    NSMutableDictionary *breakdownDictionary = nil;
    if (purchaseUnit.amount.breakdown) {
        NSDictionary *breakdownItemTotalDictionary = @{
          @"currency_code": @"USD",
          @"value": purchaseUnit.amount.breakdown.itemTotal.value
        };

        NSMutableDictionary *taxDictionary = nil;
        if (purchaseUnit.amount.breakdown.taxTotal) {
            taxDictionary = [[NSMutableDictionary alloc] initWithDictionary:@{
                @"currency_code": @"USD",
                @"value": purchaseUnit.amount.breakdown.taxTotal.value
            }];
        }

        breakdownDictionary = [[NSMutableDictionary alloc] initWithDictionary:@{
            @"item_total": breakdownItemTotalDictionary,
        }];
        
        if (taxDictionary) {
            [breakdownDictionary setValue:taxDictionary forKey:@"tax_total"];
        }
    }
        
      NSMutableDictionary *purchaseUnitAmountDictionary = [[NSMutableDictionary alloc] initWithDictionary:@{
          @"currency_code": @"USD",
          @"value": purchaseUnit.amount.value,
      }];

    if (breakdownDictionary) {
        [purchaseUnitAmountDictionary setValue:breakdownDictionary forKey:@"breakdown"];
    }
    
    NSDictionary *purchaseUnitDictionary = @{
      @"amount": purchaseUnitAmountDictionary,
      @"items": itemsDictionary
    };

    [purchaseUnitDictionaries addObject:purchaseUnitDictionary];
  }
  
  return @{
    @"intent": intentString,
    @"purchase_units": purchaseUnitDictionaries
  };
}
```

Reply:
Yep I think this is good! Thank you very much!!!!! 
I also added a check for `purchaseUnit.items`, if it's nil then we don't need to add `@"items"` in `purchaseUnitDictionary`.